src/message.rs

```diff
@@ -36,14 +36,160 @@ impl Deserialize for Version {
     }
 }
 
+/// An RPC method params
+///
+/// Not all json is a valid RPC params.
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+#[serde(untagged)]
+pub enum Params {
+    Positional(Vec<Value>),
+    Named(serde_json::Map<String, Value>),
+}
+
+impl Params {
+    /// Wether or not there are any params
+    pub fn is_empty(&self) -> bool {
+        match *self {
+            Params::Positional(ref args) => args.is_empty(),
+            Params::Named(ref fields) => fields.is_empty(),
+        }
+    }
+
+    /// Returns a json value consuming the params in the process
+    pub fn into_value(self) -> Value {
+        match self {
+            Params::Positional(xs) => Value::Array(xs),
+            Params::Named(x) => Value::Object(x),
+        }
+    }
+
+    /// Force the params to be represented in positional form
+    pub fn into_positional(self) -> Self {
+        match self {
+            Params::Named(x) => {
+                let values = x.into_iter()
```

Comment:
Damn... I figured thats why they had a Map instead of using HashMap. I'll have to find another solution for it. 

The use is for sending the rpc params in positional form instead of named.